packages/vite/src/node/plugins/metadataCompat.ts

```diff
@@ -0,0 +1,59 @@
+import type { OutputAsset, OutputChunk, RenderedChunk } from 'rollup'
+import type { ChunkMetadata, MetadataManager } from '../metadata'
+import type { Plugin } from '../plugin'
+import type { ResolvedConfig } from '..'
+
+/**
+ * Adds legacy `viteMetadata` to our new internal metadata manager.
+ * This is to support third-party plugins that rely on this property.
+ */
+export function metadataCompatPlugin(config: ResolvedConfig): Plugin {
```

Comment:
Not sure if 3rd-party plugins rely on `viteMetadata`, so added this plugin to capture any usages of it.